pennylane/devices/default_qutrit_mixed.py

```diff
@@ -0,0 +1,231 @@
+            The default, ``seed="global"`` pulls a seed from NumPy's global generator. ``seed=None``
+            will pull a seed from the OS entropy.
+            If a ``jax.random.PRNGKey`` is passed as the seed, a JAX-specific sampling function using
+            ``jax.random.choice`` and the ``PRNGKey`` will be used for sampling rather than
+            ``numpy.random.default_rng``.
+    **Example:**
+    .. code-block:: python
+
+        n_wires = 5
+        num_qscripts = 5
+        qscripts = []
+        for i in range(num_qscripts):
+            unitary = scipy.stats.unitary_group(dim=3**n_wires, seed=(42 + i)).rvs()
+            op = qml.QutritUnitary(unitary, wires=range(n_wires))
+            qs = qml.tape.QuantumScript([op], [qml.expval(qml.GellMann(0, 3))])
+            qscripts.append(qs)
+
+    >>> dev = DefaultQutritMixed()
+    >>> program, execution_config = dev.preprocess()
+    >>> new_batch, post_processing_fn = program(qscripts)
+    >>> results = dev.execute(new_batch, execution_config=execution_config)
+    >>> post_processing_fn(results)
+    [0.08015701503959313,
+    0.04521414211599359,
+    -0.0215232130089687,
+    0.062120285032425865,
+    -0.0635052317625]
+
+    This device currently supports backpropagation derivatives:
+    >>> from pennylane.devices import ExecutionConfig
+    >>> dev.supports_derivatives(ExecutionConfig(gradient_method="backprop"))
+    True
+    For example, we can use jax to jit computing the derivative:
+    .. code-block:: python
+        import jax
+        @jax.jit
+        def f(x):
+            qs = qml.tape.QuantumScript([qml.TRX(x, 0)], [qml.expval(qml.GellMann(0, 3))])
+            program, execution_config = dev.preprocess()
+            new_batch, post_processing_fn = program([qs])
+            results = dev.execute(new_batch, execution_config=execution_config)
+            return post_processing_fn(results)
+    >>> f(jax.numpy.array(1.2))
+    DeviceArray(0.36235774, dtype=float32)
+    >>> jax.grad(f)(jax.numpy.array(1.2))
+    DeviceArray(-0.93203914, dtype=float32, weak_type=True)
+    .. details::
+        :title: Tracking
+        ``DefaultQutritMixed`` tracks:
+        * ``executions``: the number of unique circuits that would be required on quantum hardware
+        * ``shots``: the number of shots
+        * ``resources``: the :class:`~.resource.Resources` for the executed circuit.
+        * ``simulations``: the number of simulations performed. One simulation can cover multiple QPU executions, such as for non-commuting measurements and batched parameters.
+        * ``batches``: The number of times :meth:`~.execute` is called.
+        * ``results``: The results of each call of :meth:`~.execute`
+    """
+
+    _device_options = ("rng", "prng_key")  # tuple of string names for all the device options.
+
+    @property
+    def name(self):
+        """The name of the device."""
+        return "default.qutrit.mixed"
```

Comment:
I'm personally not a big fan of having three words for the device name. @trbromley this is more so a question for you, but what are your thoughts on `qutrit.mixed`? It breaks away from the `default` keyword we have in our native devices so I can understand if we want to stick with `default.qutrit.mixed`. Just wanted to start a discussion.

Reply:
Yeah I wasn't sure, I just tried to continue with convention, but I agree and it looks bad in the docs.

Reply:
The only thing I'm wondering is should I replace this everywhere or just in the name attribute. I'm a bit worried about having a `devices.qutrit_mixed` module and `qml.devices.qutrit_mixed.py` file.

Reply:
Yeah that makes sense. I would say changing the name everywhere makes sense, and you can rename `default_qutrit_mixed.py` to `qutrit_mixed_device.py` or something along those lines. With that being said, I would suggest we wait on changing names until @trbromley gives some input. He is our product manager.

Reply:
Nice idea @mudit2812! The only thing to be aware of is `default.qutrit` already existing :thinking: With that in mind, I find the `default.qutrit` and `default.qutrit.mixed` pairing to be quite natural. I wonder if having `qutrit.mixed` will make users think of trying to use `qutrit` or `qutrit.pure` devices?

Reply:
Good point. In that case let's stick with `default.qutrit.mixed`.

Reply:
Yep, agreed to stick with `default.qutrit.mixed` in this PR :+1: